ip_country/src/country_finder.rs

```diff
@@ -0,0 +1,301 @@
+            &COUNTRY_CODE_FINDER_TEST,
+            IpAddr::from_str("0:0:5:0:0:0:0:0").unwrap(),
+        );
+
+        assert_eq!(result, None)
+    }
+
+    #[test]
+    fn real_test_ipv4_with_google() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("142.250.191.132").unwrap(), // dig www.google.com A
+        )
+        .unwrap();
+
+        assert_eq!(result.name, "United States".to_string());
+    }
+
+    #[test]
+    fn real_test_ipv4_with_cz_isp() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("77.75.77.222").unwrap(), // dig www.seznam.cz A
+        )
+        .unwrap();
+
+        assert_eq!(result.free_world, true);
+        assert_eq!(result.iso3166, "CZ".to_string());
+        assert_eq!(result.name, "Czechia".to_string());
+    }
+
+    #[test]
+    fn real_test_ipv4_with_sk_isp() {
+        let timestart = SystemTime::now();
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("213.81.185.100").unwrap(), // dig www.zoznam.sk A
+        )
+        .unwrap();
+        let timeend = SystemTime::now();
+
+        assert_eq!(result.free_world, true);
+        assert_eq!(result.iso3166, "SK".to_string());
+        assert_eq!(result.name, "Slovakia".to_string());
+        let duration = timeend.duration_since(timestart).unwrap().as_secs();
+        println!("duration: {}", duration);
+        assert!(duration <= 5);
+    }
+
+    #[test]
+    fn real_test_ipv6_with_google() {
+        let timestart = SystemTime::now();
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("2607:f8b0:4009:814::2004").unwrap(), // dig www.google.com AAAA
+        )
+        .unwrap();
+        let timeend = SystemTime::now();
+
+        assert_eq!(result.name, "United States".to_string());
+        let duration = timeend.duration_since(timestart).unwrap().as_secs();
+        println!("duration: {}", duration);
+        assert!(duration <= 5);
+    }
+
+    #[test]
+    fn deserialize_country_blocks_ipv4_and_ipv6_adn_fill_vecs() {
+        let mut result_ipv4: Vec<CountryBlock> = vec![];
+        let mut result_ipv6: Vec<CountryBlock> = vec![];
+        let timestart = SystemTime::now();
+        let mut deserializer_ipv4 =
+            CountryBlockDeserializerIpv4::new(crate::dbip_country::ipv4_country_data());
+        let mut deserializer_ipv6 =
+            CountryBlockDeserializerIpv6::new(crate::dbip_country::ipv6_country_data());
+        let timeend = SystemTime::now();
+
+        let timestart_fill = SystemTime::now();
+        loop {
+            match deserializer_ipv4.next() {
+                None => break, // this line isn't really testable, since the deserializer will produce CountryBlocks for every possible address
```

Comment:
Such a comment doesn't belong into a test. 

Also...it is testable, as it is the only condition at which you can leave the loop...